tendermint/src/lite/types.rs

```diff
@@ -71,19 +91,22 @@ pub trait Validator {
 /// It has an underlying Vote type with the relevant vote data
 /// for verification.
 pub trait Commit {
-    type Vote: Vote;
-
     /// Hash of the header this commit is for.
     fn header_hash(&self) -> Hash;
 
-    /// Return the underlying votes for iteration.
-    /// All votes here are for the correct block id -
-    /// we ignore absent votes and votes for nil here.
-    /// NOTE: we may want to check signatures for nil votes,
-    /// and thus use an ternary enum here instead of the binary Option.
-    // TODO figure out if we want/can do an iter() method here that returns a
-    // VoteIterator instead of returning a vec
-    fn into_vec(&self) -> Vec<Option<Self::Vote>>;
+    /// Compute the voting power of the validators that correctly signed the commit,
+    /// have according to their voting power in the passed in validator set.
+    /// Will return an error in case an invalid signature was included.
+    ///
+    /// This method corresponds to the (pure) auxiliary function int the spec:
+    /// `votingpower_in(signers(h.Commit),h.Header.V)`.
+    fn voting_power_in<V>(&self, vals: &V) -> Result<u64, Error>
```

Comment:
Now that we do this, can't we remove the `Validator` and `Vote` traits all together? They don't seem to be used in this package anymore (just in tests ...) !

Reply:
Vote can be removed now (done in https://github.com/interchainio/tendermint-rs/pull/84/commits/d907da622afdf5689f35e8d11d9e702419964f5a) 
Validator is currently used via `ValidatorSet`'s associated type, namely because of `fn validator(&self, val_id: Id) -> Option<Self::Validator>;`